tempto-core/src/main/java/com/teradata/tempto/util/Lazy.java

```diff
@@ -37,4 +39,9 @@ public synchronized T get()
         }
         return instance;
     }
+
+    public Optional<T> lazyGet()
```

Comment:
synchronized -- this class is thread-safe and guaranteed happens-before between producing value and consuming it